actors/market/tests/on_miner_sectors_terminate.rs

```diff
@@ -43,28 +44,48 @@ fn terminate_multiple_deals_from_multiple_providers() {
         .collect::<Vec<DealID>>()
         .try_into()
         .unwrap();
-    activate_deals(&rt, sector_expiry, PROVIDER_ADDR, current_epoch, &[deal1, deal2, deal3]);
+    activate_deals_legacy(&rt, sector_expiry, PROVIDER_ADDR, current_epoch, &[deal1, deal2, deal3]);
```

Comment:
For most of the tests here I think it would be better to activate deals without the legacy hacks, and keep the test clean. Invoke manual settlement where needed to progress the state.

I think this goes for many tests in other files too. We need a few tests that clearly target the transient state for legacy deals that missed their cron, but most other tests should be the new way. This will often mean replacing a cron call with manual settlement, but hopefully clarify the tests quite a bit too.

Reply:
I've kept these tests as they protect the on going behaviour for deals that are automatically settled by cron. All tests that use the legacy hack can eventually be removed. 

These are separate to the transient state deals that are marked-for-termination but not yet cleaned up - these also deserve targeted tests.

I'll write new tests (mirroring the bulk of these legacy tests) that use manual settlement to progress state.

Reply:
https://github.com/filecoin-project/builtin-actors/issues/1389

Reply:
Ok. As a suggestion to keep files manageable, maybe put the new-but-similar tests into different files so we can just delete whole legacy file when the conditions are met.